src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs

```diff
@@ -288,5 +289,23 @@ private void CredentialsSuccessfullyUsed(Uri uri, ICredentials credentials)
         {
             HttpHandlerResourceV3.CredentialsSuccessfullyUsed?.Invoke(uri, credentials);
         }
+
+        protected override void Dispose(bool disposing)
```

Comment:
Do we ever call dispose on the handler?

The thing about the dispose pattern is that it needs to be implement all the way up the stack. Are we doing that?

Reply:
It seems no.

`HttpSourceAuthenticationHandler` objects are created inside `HttpHandlerResourceV3Provider.TryCreate()` method. Objects returned by this method are `INuGetResource`-based. This interface does not implement any `IDisposable`-based interfaces.

Reply:
In short words: we don't call dispose on this class in NuGet.Client source code. We are now just being good citizens by implementing Dispose.

Reply:
How about we just accept this `dispose` changes as it's. But create/link another follows up issue for implementing dispose pattern all the way up the stack, fixing that part shouldn't be high priority for now. 
Or just suppress with reason why it doesn't add any value by doing so.

Reply:
I'm looking forward to merging this PR. Added follow up issue to use IDisposable pattern in NuGet protocol code. https://github.com/NuGet/Home/issues/12176

Reply:
Should we call `GC.SuppressFinalize(this);`  in dispose? We're doing it for another dispose.
https://github.com/NuGet/NuGet.Client/blob/10ce64e63bb15f5fed2f39aa129f70fb370d106a/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetSourcesService.cs#L89

Reply:
Not here, that should be called in Dispose() method (without arguments) of base class

Reply:
> Not here, that should be called in Dispose() method (without arguments) of base class
Sure, thank you for clarifying.